qa/tasks/cephfs/test_multifs.py

```diff
@@ -0,0 +1,421 @@
+
+from teuthology.misc import sudo_write_file
+from teuthology.orchestra.run import CommandFailedError
+from teuthology.orchestra.run import Raw
+
+
+log = logging.getLogger(__name__)
+
+
+class MountDetails():
+
+    def __init__(self, client_id, client_keyring_path, client_remote,
+                 cephfs_name, cephfs_mntpt, hostfs_mntpt):
+        self.client_id = client_id
+        self.client_keyring_path = client_keyring_path
+        self.client_remote = client_remote
+        self.cephfs_name = cephfs_name
+        self.cephfs_mntpt = cephfs_mntpt
+        self.hostfs_mntpt = hostfs_mntpt
+
+    def restore(self, mntobj):
+        mntobj.client_id = self.client_id
+        mntobj.client_keyring_path = self.client_keyring_path
+        mntobj.client_remote = self.client_remote
+        mntobj.cephfs_name = self.cephfs_name
+        mntobj.cephfs_mntpt = self.cephfs_mntpt
+        mntobj.hostfs_mntpt = self.hostfs_mntpt
+
+
+class TestMultiFS(CephFSTestCase):
+    client_id = 'testuser'
+    client_name = 'client.' + client_id
+    # one dedicated for each FS
+    MDSS_REQUIRED = 2
+    CLIENTS_REQUIRED = 2
+
+    def setUp(self):
+        super(TestMultiFS, self).setUp()
+
+        # XXX: saving mount details that were created by CephFSTestCase.setUp()
+        # so that these details are restored later to ensure smooth teardown
+        # for upcoming tests.
+        self.mount_details = [
+            MountDetails(
+                self.mount_a.client_id, self.mount_a.client_keyring_path,
+                self.mount_a.client_remote, self.mount_a.cephfs_name,
+                self.mount_a.cephfs_mntpt, self.mount_a.hostfs_mntpt),
+            MountDetails(
+                self.mount_b.client_id, self.mount_b.client_keyring_path,
+                self.mount_b.client_remote, self.mount_b.cephfs_name,
+                self.mount_b.cephfs_mntpt, self.mount_b.hostfs_mntpt)]
+
+        self.fs1 = self.fs
+        self.fs1.enable_multifs()
+        self.fs2 = self.mds_cluster.newfs(name='cephfs2', create=True)
+
+        # we'll reassign caps to client.1 so that it can operate with cephfs2
+        self.run_cluster_cmd(f'auth caps client.{self.mount_b.client_id} mon '
+                             f'"allow r" osd "allow rw '
+                             f'pool={self.fs2.get_data_pool_name()}" mds allow')
+        self.mount_b.remount(cephfs_name=self.fs2.name)
+
+    def tearDown(self):
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+
+        self.mount_details[0].restore(self.mount_a)
+        self.mount_details[1].restore(self.mount_b)
+
+    def create_keyring_file(self, remote, keyring):
+        keyring_path = remote.run(args=['mktemp'], stdout=StringIO()).\
+            stdout.getvalue().strip()
+        sudo_write_file(remote, keyring_path, keyring)
+
+        # required when triggered using vstart_runner.py.
+        remote.run(args=['chmod', '777', keyring_path])
```

Comment:
running in vstart requires `0777`? Shouldn't it be something like `644`?

Reply:
Yes, that should work. Fixed.